lib/matplotlib/offsetbox.py

```diff
@@ -1339,27 +1341,18 @@ def __init__(self, offsetbox, xy,
         """
 
         martist.Artist.__init__(self)
-        mtext._AnnotationBase.__init__(self,
-                                       xy,
-                                       xycoords=xycoords,
-                                       annotation_clip=annotation_clip)
+        mtext._AnnotationBase.__init__(
+            self, xy, xycoords=xycoords, annotation_clip=annotation_clip)
 
         self.offsetbox = offsetbox
-
-        self.arrowprops = arrowprops
+        self._box_alignment = box_alignment
 
         self.set_fontsize(fontsize)
 
-        if xybox is None:
-            self.xybox = xy
-        else:
-            self.xybox = xybox
-
-        if boxcoords is None:
-            self.boxcoords = xycoords
-        else:
-            self.boxcoords = boxcoords
+        self.xybox = xybox if xybox is not None else xy
+        self.boxcoords = boxcoords if boxcoords is not None else xycoords
```

Comment:
```suggestion
        self.xybox = xybox or xy
        self.boxcoords = boxcoords or xycoords
```

Reply:
The problem of `or` is if someone passes the coords as a size-2 numpy array, in which case you error out with `The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`

Reply:
Ahh, yes, forgot about that one! Thanks for clarifying.